test/Frontend/module-alias-import-attrs.swift

```diff
@@ -0,0 +1,190 @@
+
+/// Test various import attributes with module aliasing.
+///
+/// Module 'Lib' imports module 'XLogging' via module aliasing and with various import attributes.
+/// Module 'User' imports 'Lib'.
+
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
+
+/// Create AppleLogging.swiftmodule by aliasing XLogging
+// RUN: %target-swift-frontend -module-name AppleLogging -module-alias XLogging=AppleLogging %t/FileLogging.swift -emit-module -emit-module-path %t/AppleLogging.swiftmodule
+// RUN: test -f %t/AppleLogging.swiftmodule
+
+/// Below, compile module Lib that imports XLogging with various import attributes via
+/// -module-alias XLogging=AppleLogging
+
+/// Test @_spi: Should fail
+// RUN: not %target-swift-frontend -module-name Lib1 %t/FileLib1.swift -module-alias XLogging=AppleLogging -I %t -emit-module -emit-module-path %t/Lib1.swiftmodule 2> %t/result-Lib1.output
+// RUN: %FileCheck %s -input-file %t/result-Lib1.output -check-prefix CHECK-SPI-ERROR
+// CHECK-SPI-ERROR: error: 'spiFunc' is inaccessible due to '@_spi' protection level
+
+/// Test @_spi: Should pass by adding @_spi
+// RUN: %target-swift-frontend -module-name Lib2 %t/FileLib2.swift -module-alias XLogging=AppleLogging -I %t -emit-module -emit-module-path %t/Lib2.swiftmodule
+// RUN: test -f %t/Lib2.swiftmodule
+// RUN: llvm-bcanalyzer -dump %t/Lib2.swiftmodule > %t/Lib2.dump.txt
+// RUN: %FileCheck -check-prefix=CHECK-REAL-NAME %s < %t/Lib2.dump.txt
+// CHECK-REAL-NAME: AppleLogging
+// RUN: not %FileCheck -check-prefix=CHECK-ALIAS %s < %t/Lib2.dump.txt
+// CHECK-ALIAS: XLogging
```

Comment:
I would recommend using the `-NOT` approach everywhere for negative checks. A `not FileCheck` test may be satisfied by other errors, like if a future modification uses an invalid argument here.

Reply:
👍